test/components/connectors/test_openapi_service.py

```diff
@@ -13,14 +17,30 @@ def openapi_service_mock():
 
 
 @pytest.fixture
-def service_auths():
-    return {"TestService": "auth_token"}
+def random_open_pull_request_head_branch() -> str:
+    token = os.getenv("GITHUB_TOKEN")
+    headers = {"Accept": "application/vnd.github.v3+json", "Authorization": f"token {token}"}
+    response = requests.get("https://api.github.com/repos/deepset-ai/haystack/pulls?state=open", headers=headers)
+
+    if response.status_code == 200:
+        pull_requests = response.json()
+        for pr in pull_requests:
+            if pr["base"]["ref"] == "main":
+                return pr["head"]["ref"]
+    else:
+        raise Exception(f"Failed to fetch pull requests. Status code: {response.status_code}")
+
+
+@pytest.fixture
+def genuine_fc_message(random_open_pull_request_head_branch):
+    basehead = "main..." + random_open_pull_request_head_branch
+    return f"""[{{"id": "call_NJr1NBz2Th7iUWJpRIJZoJIA", "function": {{"arguments": "{{\\n  \\"parameters\\": {{\\n    \\"basehead\\": \\"{basehead}\\",\\n    \\"owner\\": \\"deepset-ai\\",\\n    \\"repo\\": \\"haystack\\"\\n  }}\\n}}", "name": "compare_branches"}}, "type": "function"}}]"""
```

Comment:
Can't we use `json.dumps()` here instead of writing the JSON as string?

Reply:
Sure, I wanted to have the original message as is but it the same thing.....